### DISCUSSION TITLE
How can I place Likert-Items in a matrix/table-style arrangement?

### DISCUSSION THREAD

**Discussion by slussinator** [ctreffe/alfred#182] [Alfred Q&A] [answered]
I would like to present a questionnaire (60 items) on alfred. 

First, I would like the items to be flush-left and the choices to appear next to each item. The choices "keys" should be represented at the top of the page, not directly next to the choice button. I have attached a picture of what I mean. 
Is there an easy way to do this?

Secondly, I would like only ten items to be shown on each page, such that the questionnaire would have 6 pages. My items are saved in a content.py and I don't want to have to write each item in my script.py. 
I know how to get 1 item per page / all items on one page, but not how to combine a subset of items. 
Is this possible?

I hope this makes sense. Thank you in advance!

<img width="492" alt="Bildschirmfoto 2022-03-09 um 15 22 39" src="https://user-images.githubusercontent.com/91884644/157460430-e5edbc48-6576-45c4-849d-a6ba57f434fa.png">

**Comment by jobrachem** [comment, marked as answer]
## Left-aligned label

You can have your label left-aligned with the help of the `al.Label` class:

```python
import alfred3 as al
exp = al.Experiment()

@exp.member
class Demo(al.Page):

    def on_exp_access(self):
        labels = ["choice1", "choice2", "choice3", "choice4", "choice5"]
        self += al.SingleChoiceButtons(
            *labels,
            leftlab=al.Label("My item label", align="left"),
            name="demo"
        )
```

Results in:
<img width="772" alt="image" src="https://user-images.githubusercontent.com/37882800/158023874-e641edfd-f57e-4b37-9703-58f02ad44b4c.png">

## Matrix-Style presentation

You can construct something similar to the classic matrix-style arrangement using `al.ButtonLabels`. Note that you need to set `leftlab=" "` (with a space) to achieve the correct alignment. I also set `button_style="btn-secondary"` to make the labels visually distinct form actual clickable buttons. You can use the `layout` argument of the `al.SingleChoiceButtons` and `al.ButtonLabels`, if you need more horizontal space for your left label. Or you can use an `al.WidePage` as the page base class to make the whole page wider.

```python
import alfred3 as al
exp = al.Experiment()

@exp.member
class Demo(al.Page):

    def on_exp_access(self):
        labels = ["choice1", "choice2", "choice3", "choice4", "choice5"]
        
        self += al.ButtonLabels(*labels, leftlab=" ", button_style="btn-secondary")
        
        self += al.SingleChoiceButtons(
            *range(1, 6),
            leftlab=al.Label("My item label", align="left"),
            name="demo"
        )
```

<img width="764" alt="image" src="https://user-images.githubusercontent.com/37882800/158024104-9c000c4c-984e-434f-af1c-267822e44c75.png">

## Subset of items

Regarding your third question, I think it would be most useful to open a new, dedicated question. That will make it easier for other users to find the question. Also, the question becomes easier to answer, if you include a minimal experiment of your solution for one page per item 😊 . I think we will be able to achieve your desired result by cleverly using list subsetting, i.e., something similar to the following:

```python
>>> my_list = ["a", "b", "c", "d", "e", "f", "g"]
>>> my_list[0:3]
['a', 'b', 'c']
```

For a full answer with a minimal working alfred experiment, please open a new question 👍

**Reply by slussinator** [reply]
Thank you very much! This was quite helpful. I will open a new question with minimal working code!
